### PR TITLE
Pilot 5588: update cli logic to adopt Windows system

### PR DESCRIPTION
**Pull request by colorzzr**
## Summary

with new requirement, now cli will also be executed on Windows machine, which lead to a requirement to test code base with windows adaptability. This PR made following updates to achieve such requirements:
 1. update github actions to build cli binary in windows runner.
 2. update `file upload` command to convert windows path(eg. folder\\file.txt) to object storage path(eg. folder/file.txt).
 3. update poetry file with `extras` section for windows packages.
 4. update configuration file permission with Windows permissions.

https://indocconsortium.atlassian.net/wiki/spaces/PILOT/pages/3913056332/Permission+of+Cli+Configurations+ADR

## JIRA Issues

PILOT-5588

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Refactor or reformatting

## Testing

Are there any new or updated tests to validate the changes?

- [x] Yes
- [ ] No

## Test Directions

fixup test cases with new updates

### CODE REVIEW

File: app/configs/user_config.py [hunk 1]
```diff
@@ -51,11 +55,12 @@ def __init__(
...
-            config_path.mkdir(mode=0o0700, exist_ok=False)
+            if platform.system() == 'Windows':
+                create_directory_with_permissions_windows(config_path)
+            else:
+                config_path.mkdir(mode=0o0700, exist_ok=False)
 
-        current_user_id = os.geteuid()
-
-        error = self._check_user_permissions(config_path, current_user_id, (0o0500, 0o0700))
+        error = self._check_user_permissions(config_path, (0o0500, 0o0700))
```
Comment (erikdvlp) [inline app/configs/user_config.py]: Do you think we should have enum variants for these octal permissions for readability?

**Review by erikdvlp — APPROVED** [review]

File: app/configs/user_config.py (hunk 1 above)
Comment (colorzzr) [reply app/configs/user_config.py]: Oh yes `Enum` variables give better visual representations. Will update them!

File: app/configs/user_config.py [hunk 2]
```diff
@@ -49,22 +64,30 @@ def __init__(
             is_bundled = getattr(sys, 'frozen', False) and hasattr(sys, '_MEIPASS')
             is_cloud_mode = is_bundled and (Path(sys._MEIPASS) / 'ENABLE_CLOUD_MODE').is_file()
 
+        # print(int(FilePermissions.READ_WRITE_EXECUTE_USER))
+        # raise
+
```
Comment (erikdvlp) [inline app/configs/user_config.py]:
```suggestion
```

File: .github/workflows/build-and-publish.yml [hunk 3]
```diff
@@ -145,3 +145,56 @@ jobs:
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.extract-branch-name.outputs.branch }}
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
```
Comment (andricDu) [inline .github/workflows/build-and-publish.yml]: should we update this to 3.10 now?

**Review by andricDu — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-publish.yml (hunk 3 above)
Comment (colorzzr) [reply .github/workflows/build-and-publish.yml]: oh yes will update to 3.10.12

**Review by andricDu — APPROVED** [review]

**Pull request merged**